src/protocols/aeternity/libs/AeAccountHdWallet.ts

```diff
@@ -89,6 +94,96 @@ export class AeAccountHdWallet extends MemoryAccount {
     );
   }
 
+  override async signTypedData(
+    data: Encoded.ContractBytearray,
+    aci: Parameters<AccountBase['signTypedData']>[1],
+    options: Parameters<AccountBase['signTypedData']>[2] = {},
+  ): Promise<Encoded.Signature> {
+    if (options?.aeppOrigin) {
+      const dataType = new TypeResolver().resolveType(aci);
+      const decodedData = new ContractByteArrayEncoder().decodeWithType(data, dataType);
+      const {
+        name, version, networkId, contractAddress,
+      } = options;
+      const opt = {
+        name, version, networkId, contractAddress, aci, data, decodedData,
+      };
+      const bigintReplacer = (k: string, v: any) => (
+        typeof v === 'bigint' ? `${v} (as BigInt)` : v
+      );
+      const message = `sign typed data:\n${JSON.stringify(opt, bigintReplacer, 2)}`;
+      const { checkOrAskPermission } = usePermissions();
+      const permissionGranted = await checkOrAskPermission(
+        METHODS.signTypedData,
+        options.aeppOrigin,
+        { message },
+      );
+      if (!permissionGranted) {
+        throw new RpcRejectedByUserError();
+      }
+    }
+
+    return super.signTypedData(
+      data,
+      aci,
+      {
+        ...options, // Mainly to pass the `fromAccount` property
+        aeppOrigin: undefined,
+      },
+    );
+  }
+
+  override async signDelegation(
+    delegation: Encoded.Bytearray,
+    options: Parameters<AccountBase['signDelegation']>[1] = {},
+  ): Promise<Encoded.Signature> {
+    let message;
+    let resolvedName;
+    const { getMiddleware } = useAeMiddleware();
+    if (options?.aeppOrigin) {
+      const params = unpackDelegation(delegation);
+      switch (params.tag) {
+        case DelegationTag.AensName:
+          try {
+            resolvedName = (await (await getMiddleware()).getName(params.nameId)).name;
+          } catch (e) {
+            handleUnknownError(e);
+          }
+          message = `sign delegation of ${resolvedName || params.nameId} to ${params.contractAddress}`;
+          break;
+        case DelegationTag.AensPreclaim:
+          message = `sign delegation of name preclaim to ${params.contractAddress}`;
+          break;
+        case DelegationTag.AensWildcard:
+          message = `sign delegation of name claiming to ${params.contractAddress}`;
```

Comment:
```suggestion
          message = `sign delegation of all names management to ${params.contractAddress}`;
```
> AENS_CLAIM, AENS_UPDATE, AENS_TRANSFER, AENS_REVOKE

https://github.com/aeternity/protocol/blob/8a9d1d1/contracts/fate.md#appendix-2-delegation-signatures